Modules/_sqlite/connection.c

```diff
@@ -118,11 +118,15 @@ pysqlite_connection_init(pysqlite_Connection *self, PyObject *args,
     if (!isolation_level) {
         isolation_level = PyUnicode_FromString("");
         if (!isolation_level) {
+            PyErr_SetString(pysqlite_ProgrammingError, "Isolation level could not be set.");
             return -1;
         }
     } else {
         Py_INCREF(isolation_level);
     }
+    
+    self->initialized = 1;
+    
```

Comment:
We should not set this unless `pysqlite_connection_init` actually is successful. That is at the end of this function, just before it returns `0`.